applications/ConvectionDiffusionApplication/custom_response_functions/local_temperature_average_response_function.h

```diff
@@ -164,7 +164,7 @@ class LocalTemperatureAverageResponseFunction: public AdjointResponseFunction
             GetTargetModelPart(rModelPart, mTargetModelPartName);
 
         const double domain_aggregated_temperature =
-            VariableUtils().SumHistoricalVariable(TEMPERATURE, r_target_model_part);
+            VariableUtils().SumHistoricalVariable<double>(TEMPERATURE, r_target_model_part);
```

Comment:
Shouldn't it automatically detect the type?

Reply:
I thought the same, but it gives a compiler error :/

Reply:
Can you please post the error?
I would like to know what the reason is
(Not a blocker)

Reply:
```
In file included from /media/arming/Volume/04_Development/Kratos_VW/applications/ConvectionDiffusionApplication/custom_python/add_custom_response_functions_to_python.cpp:20:0:
/media/arming/Volume/04_Development/Kratos_VW/applications/ConvectionDiffusionApplication/custom_response_functions/local_temperature_average_response_function.h: In member function ‘virtual double Kratos::LocalTemperatureAverageResponseFunction::CalculateValue(Kratos::ModelPart&)’:
/media/arming/Volume/04_Development/Kratos_VW/applications/ConvectionDiffusionApplication/custom_response_functions/local_temperature_average_response_function.h:167:83: error: no matching function for call to ‘Kratos::VariableUtils::SumHistoricalVariable(Kratos::Variable<double>&, const Kratos::ModelPart&)’
             VariableUtils().SumHistoricalVariable(TEMPERATURE, r_target_model_part);
                                                                                   ^
In file included from /media/arming/Volume/04_Development/Kratos_VW/applications/ConvectionDiffusionApplication/custom_response_functions/local_temperature_average_response_function.h:17:0,
                 from /media/arming/Volume/04_Development/Kratos_VW/applications/ConvectionDiffusionApplication/custom_python/add_custom_response_functions_to_python.cpp:20:
/media/arming/Volume/04_Development/Kratos_VW/kratos/utilities/variable_utils.h:917:15: note: candidate: template<class TDataType, class TVarType> TDataType Kratos::VariableUtils::SumHistoricalVariable(const TVarType&, const Kratos::ModelPart&, unsigned int)
     TDataType SumHistoricalVariable(
               ^~~~~~~~~~~~~~~~~~~~~
/media/arming/Volume/04_Development/Kratos_VW/kratos/utilities/variable_utils.h:917:15: note:   template argument deduction/substitution failed:
In file included from /media/arming/Volume/04_Development/Kratos_VW/applications/ConvectionDiffusionApplication/custom_python/add_custom_response_functions_to_python.cpp:20:0:
/media/arming/Volume/04_Development/Kratos_VW/applications/ConvectionDiffusionApplication/custom_response_functions/local_temperature_average_response_function.h:167:83: note:   couldn't deduce template parameter ‘TDataType’
             VariableUtils().SumHistoricalVariable(TEMPERATURE, r_target_model_part);
                                                                                   ^
```

Reply:
Thanks!